nexus/client.go

```diff
@@ -0,0 +1,177 @@
+//     Copyright 2020 Aicura Nexus Client and/or its authors
+//
+//     This file is part of Aicura Nexus Client.
+//
+//     Aicura Nexus Client is free software: you can redistribute it and/or modify
+//     it under the terms of the GNU Lesser General Public License as published by
+//     the Free Software Foundation, either version 3 of the License, or
+//     (at your option) any later version.
+//
+//     Aicura Nexus Client is distributed in the hope that it will be useful,
+//     but WITHOUT ANY WARRANTY; without even the implied warranty of
+//     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//     GNU Lesser General Public License for more details.
+//
+//     You should have received a copy of the GNU Lesser General Public License
+//     along with Aicura Nexus Client.  If not, see <https://www.gnu.org/licenses/>.
+
+package nexus
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+
+	"go.uber.org/zap"
+)
+
+const (
+	userAgent              = "acuri-nexus-client"
+	applicationJSONContent = "application/json"
+	defaultNewAPIVersion   = "beta"
+	apiPath                = "service/rest"
+)
+
+type service struct {
+	client *Client
+}
+
+// Client base structure for the Nexus Client
+type Client struct {
+	httpClient *http.Client
+	baseURL    *url.URL
+	username   string
+	password   string
+	apiVersion string
+	shared     service // same instance shared among all services
+	logger     *zap.SugaredLogger
+
+	UserService *UserService
+}
+
+// ClientBuilder fluent API to build a new Nexus Client
+type ClientBuilder struct {
+	client *Client
+}
```

Comment:
How about we don't name the client variable? It's the only variable we have on the struct, so I don't think we'd lose clarity and it'd make using the builder a bit less verbose while still allowing us to extend the type's behavior via methods, like [here](https://github.com/m88i/nexus-operator/blob/master/pkg/test/client.go#L39).

```suggestion
type ClientBuilder struct {
	*Client
}
```
And we could set its credentials, for example, like so:

```go
b.username = username
b.password = password
```

Instead of:

```go
b.client.username = username
b.client.password = password
```
Wdyt?